### ISSUE TITLE
テスト品質に関する問題点の改善

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochitrain#302]
## Target Code

- `tests/unit/test_core/test_layer_wise_lr.py`
- `pochitrain/utils/model_loading.py`
- `pochitrain/utils/json_utils.py`
- `tests/integration/`
- テスト全体のフィクスチャ設計

### 1. `test_layer_wise_lr_validation_error` のアサーション不足 (高)

`test_layer_wise_lr.py:127-138` で, テスト名が「バリデーションエラー」を示唆しているが, 実際にはエラーが発生することを検証していない. `pytest.raises` でエラーをキャッチするか, テスト名を実態に合わせて修正する必要がある.

```python
# tests/unit/test_core/test_layer_wise_lr.py:127-138
def test_layer_wise_lr_validation_error(self, trainer):
    """不正な層別学習率設定でのエラーテスト."""
    layer_wise_lr_config = {"layer_rates": {}}
    trainer.setup_training(
        learning_rate=0.001,
        optimizer_name="SGD",
        enable_layer_wise_lr=True,
        layer_wise_lr_config=layer_wise_lr_config,
    )
    assert trainer.enable_layer_wise_lr  # エラーの検証がない
```

### 2. `utils/model_loading.py` のテスト不足 (中)

`model_loading.py` に専用テストファイルがない. 特に `TypeError` フォールバック処理のテストが必要.

### 3. `utils/json_utils.py` のテスト不足 (中)

`json_utils.py` にテストファイルが存在しない.

### 4. 統合テストの不足 (中)

`tests/integration/` に統合テストが3件のみで, エンドツーエンドの訓練フローテストがない.

- `model_loading.py` の `TypeError` フォールバック処理テスト
- `json_utils.py` のユニットテスト追加
- `test_layer_wise_lr_validation_error` の修正
- 統合テストの充実 (訓練フローのスモークテスト等)

### 5. フィクスチャの重複 (低)

`test_training_configurator.py` と `test_layer_wise_lr.py` で類似のフィクスチャが重複. 共通 conftest への集約を検討.

### 6. `tempfile.TemporaryDirectory` と `tmp_path` の混在 (低)

テスト全体で一時ディレクトリの作成方法が統一されていない.

## Acceptance Criteria

- `test_layer_wise_lr_validation_error` が適切にエラーを検証する, またはテスト名が実態に合っている
- `model_loading.py` の専用テストが追加されている
- `json_utils.py` の専用テストが追加されている
- 既存テストが全てパスする
- テストカバレッジが低下していない

## Labels

`test`

## Branch

`test/improve-test-quality`
